on_device_training/mobile/ios/MyVoice/InferView.swift

```diff
@@ -0,0 +1,113 @@
+            }
+            endRecord(recognizeResult)
+        }
+    }
+    
+    private func endRecord(_ result: Result<(Bool, Float), Error>) {
+        DispatchQueue.main.async {
+            switch result {
+            case .success(let (isMatch, confidence)):
+                print("Your Voice with confidence: \(isMatch),  \(confidence)")
+                inferResult = isMatch ? .user : .other
+                probUser = confidence
+            case .failure(let error):
+                print("Error: \(error)")
+            }
+            readyToRecord = true
+        }
+    }
+    
+    var body: some View {
+        VStack {
+            Spacer()
+            
+            ZStack(alignment: .center) {
+                Image(systemName: "mic.fill")
+                    .resizable()
+                    .aspectRatio(contentMode: .fit)
+                    .frame(width: 100, height: 100)
+                    .foregroundColor( readyToRecord ? .gray: .red)
+                    .transition(.scale)
+                    .animation(.easeInOut, value: 1)
+            }
+            
+            Spacer()
+            
+            Button(action: {
+                readyToRecord = false
+                recordVoice()
+            }) {
+                Text(readyToRecord ? "Record" : "Recording ...")
+                    .font(.title)
+                    .padding()
+                    .background(readyToRecord ? .green : .gray)
+                    .foregroundColor(.white)
+                    .cornerRadius(10)
+                
+            }.disabled(isErrorOccurred || !readyToRecord)
+                .opacity(isErrorOccurred ? 0.5: 1.0)
+            
+            if  inferResult != .notSet {
+                Spacer()
+                ZStack (alignment: .center) {
+                    Image(systemName: inferResult == .user ? "person.crop.circle.fill.badge.checkmark": "person.crop.circle.fill.badge.xmark")
+                        .resizable()
+                        .aspectRatio(contentMode: .fit)
+                        .frame(width: 100, height: 100)
+                        .foregroundColor(inferResult == .user ? .green : .red)
+                        .animation(.easeInOut, value: 2)
+                    
+                }
+                
+                Text("Probability of User : \(String(format: "%.2f", probUser*100.0))%")
+                    .multilineTextAlignment(.center)
+                    .fontDesign(.monospaced)
+            }
+            
+            Spacer()
+        }
+        .padding()
+        .navigationTitle("Infer")
+        .onAppear {
+            do {
+                voiceIdentifier = try  VoiceIdentifier()
+                
+            } catch {
+                alertMessage = "Error initializing inference session, make sure that training is completed: \(error)"
+                showAlert = true
+                isErrorOccurred = true
```

Comment:
consider a name like `hasInitializationErrorOccurred`. alternatively, could we just check if `voiceIdentifier != nil`?